packages/multicast_dns/lib/src/lookup_resolver.dart

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2015, the Dartino project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:collection';
+
+import 'package:multicast_dns/src/resource_record.dart';
+
+/// Class for maintaining state about pending mDNS requests.
+class PendingRequest extends LinkedListEntry<PendingRequest> {
+  /// Creates a new PendingRequest.
+  PendingRequest(this.type, this.name, this.controller);
+
+  /// The [ResourceRecordType] of the request.
+  final int type;
+
+  /// The domain name.
```

Comment:
Maybe a little more detail?  Is this the domain name of the host that the request is going to? Is it just the domain, or is it a full hostname? (e.g. flutter.io vs docs.flutter.io)